### PR TITLE
chore: new flat left navigation panel (all language tracks)

### PR DESCRIPTION
**Pull request by wardbell**
Updated version of #856 

Todos:
* [x] @naomiblack to review & approve
* [x] DEVELOPER GUIDE: add Overview page, derived from existing DevGuide index.html (Ward to wordsmith)
* [x] TESTING: consolidate testing chapters and fold into DevGuide(Ward)
* [ ] Add node-based broken link checker and tie it into CI

### CODE REVIEW

File: public/docs/dart/latest/_nav.jade [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+      li: a(href='#{base}guide/dependency-injection.html') Dependency Injection
+      li: a(href='#{base}guide/template-syntax.html') Template Syntax
+      li: a(href='#{base}guide/cheatsheet.html') Angular Cheat Sheet        
+.nav-blocks
+  .nav-title Developer Guide
+  .nav-unordered-lists
+    ul
+      li: a(href='#{base}guide/attribute-directives.html') Attribute Directives
+      li: a(href='#{base}guide/hierarchical-dependency-injection.html') Hierarchical Injectors
+      li: a(href='#{base}guide/server-communication.htm') Http Client
```
Comment (kwalrath) [inline public/docs/dart/latest/_nav.jade]: htm -> html

File: public/docs/dart/latest/_nav.jade [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+      li: a(href='#{base}guide/lifecycle-hooks.html') Lifecycle Hooks
+      li: a(href='#{base}guide/pipes.html') Pipes
+      li: a(href='#{base}guide/router.html') Routing and Navigation
+      li: a(href='#{base}guide/structural-directives.html') Structural Directives
+      li: a(href='#{base}guide/upgrade.html') Upgrading from 1.x
```
Comment (kwalrath) [inline public/docs/dart/latest/_nav.jade]: We don't plan to provide this for Dart. OK to delete it?

File: public/docs/dart/latest/_nav.jade [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+.nav-blocks
+  .nav-title Cookbook
+  .nav-unordered-lists
+    ul
+      li: a(href='#{base}cookbook') Overview
+      li: a(href='#{base}cookbook/a1-a2-quick-reference.html') Angular 1 to 2 Quick Ref
```
Comment (kwalrath) [inline public/docs/dart/latest/_nav.jade]: Again, we don't plan to provide this in Dart.

File: public/docs/dart/latest/_nav.jade [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+      li: a(href='#{base}cookbook/component-communication.html') Component Interaction
+.nav-blocks
+  .nav-title Testing
+  .nav-unordered-lists
+    ol
+      li: a(href='#{base}testing') Overview
+      li: a(href='#{base}testing/jasmine-testing-101.html') Jasmine Testing 101
```
Comment (kwalrath) [inline public/docs/dart/latest/_nav.jade]: I don't think this is relevant to Dart. Would removing it cause problems with other testing chapters? If so, maybe we should remove all of Testing from the Dart version, for now.

Comment (wardbell) [inline public/docs/dart/latest/_nav.jade]: Easy to remove. Look at _nav.jade under Dart folder. Hack away.

File: public/docs/dart/latest/_nav.jade [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+      li: a(href='#{base}testing/application-under-test.html') Application Under Test
+      li: a(href='#{base}testing/first-app-tests.html') First App Tests
+      li: a(href='#{base}testing/testing-an-angular-pipe.html') Testing an Angular Pipe
+.nav-blocks
+  .nav-title Reference
+  .nav-unordered-lists
+    ul
+      li: a(href='#{base}api') API Preview
+      li: a(href='#{base}cheatsheet.html') Angular Cheat Sheet
```
Comment (kwalrath) [inline public/docs/dart/latest/_nav.jade]: It's a little weird to have the cheat sheet in two places in the sidenav. I can't think of a better solution, though.

**Comment by kwalrath** [comment]
Looks good! I'll need to update the Dart quickstart's link to the developer guide (since it's no longer to the developer guide, technically). But that's not a big deal.

File: public/docs/dart/latest/_nav.jade (hunk 1 above)
Comment (wardbell) [inline public/docs/dart/latest/_nav.jade]: Fixed in all three language tracks.

File: public/docs/dart/latest/_nav.jade (hunk 4 above)
Comment (wardbell) [inline public/docs/dart/latest/_nav.jade]: Removed "Testing" from Dart and JavaScript tracks.

**Comment by wardbell** [comment]
@kwalrath You've reminded me that I need to adjust what was the DevGuide intro to be the Basics intro

**Comment by wardbell** [comment]
@kwalrath I revised the Basics overview; its now a Documentation overview. You may want to leverage it  for Dart (see the JavaScript version for clues).

**Pull request merged**